### PR TITLE
Update contributing docs

### LINKED ISSUE #1194: MEP description in docs needs to be updated
Considering most of the accepted enhancement proposals so far have been in docs, I think it's better to link to [another template in docs](https://docs.google.com/document/d/1adomheXpbqp4YIBFQ49IsAJzuJKWyr75GRO1NeWg0Fo/edit?usp=sharing). 

It would also be better to avoid the confusion between RFC and MEP by updating[ the documentation](https://mitiq.readthedocs.io/en/stable/contributing.html#proposing-a-new-feature-to-mitiq). I propose keeping both the `md` and `doc` versions just in case. 

The `md` template also has a missing [link](https://github.com/unitaryfund/mitiq/blob/master/docs/mep/0000-feature-proposal-TEMPLATE.md#references-and-footnotes) to another MEP. 

image

### PR DESCRIPTION
**Pull request by amirebrahimi**
Description
-----------

Removed old MEP and update documentation to point to the new template
Resolves #1194 

Checklist
---------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [X] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [X] Added myself / the copyright holder to the AUTHORS file

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

License
-------

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Tips
----

- If the validation check fails:

    1. Run `make check-types` (from the root directory of the repository) and fix any [mypy](https://mypy.readthedocs.io/en/stable/) errors.

    2. Run `make check-style` and fix any [flake8](http://flake8.pycqa.org) errors.

    3. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html) autoformatter.

  For more information, check the [Mitiq style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).

- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,12 +154,15 @@ default of 88 (this is configured in the [`pyproject.toml`](https://github.com/u
...
-discussion about the feature you would like to add, make a copy of the template and increment the proposal
-number and change `feature-proposal-TEMPLATE` to a short description of what you are proposing.
-Please fill out any relevant sections of that template as best you can and we can discuss in
-both PR threads as well as on the [discord](http://discord.unitary.fund).
+If you are interested in adding a larger new feature or other functionality to Mitiq, please make a copy of our
+Request For Comments (RFC) [template](https://docs.google.com/document/d/1adomheXpbqp4YIBFQ49IsAJzuJKWyr75GRO1NeWg0Fo/)
+and fill out the details of your enhancement proposal. Take a look at
+[previous RFCs](https://github.com/unitaryfund/mitiq/issues?q=is%3Aissue+is%3Aclosed+RFC) for examples on how to fill
+out your proposal. Once you have completed your proposal, then please
+[create a feature request PR](https://github.com/unitaryfund/mitiq/issues/new?assignees=&labels=feature-request&template=feature_request.md&title=)
```
Comment (andreamari) [inline CONTRIBUTING.md:162]:
```suggestion
[create a feature request issue](https://github.com/unitaryfund/mitiq/issues/new?assignees=&labels=feature-request&template=feature_request.md&title=)
```

**Review by andreamari — APPROVED** [review]
Great! I guess you mean "issue" instead of "PR", right?

**Comment by andreamari** [comment]
Fill free to add yourself to the AUTHORS file. Unless you are already in that list

**Comment by amirebrahimi** [comment]
> Fill free to add yourself to the AUTHORS file. Unless you are already in that list

Seems like a minor reason to add to the AUTHORS file, but went ahead. Hopefully my RCI change will make it in at some point.

**Comment by amirebrahimi** [comment]
Looks like we're good to merge!

**Pull request merged**